src/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/RequestExecutionQueue.cs

```diff
@@ -206,62 +238,15 @@ private async Task ProcessQueueAsync()
-                        }
 
-                        Debug.Assert(!concurrentlyExecutingTasks.Any(t => !t.Key.IsCompleted), "The tasks should have all been drained before continuing");
-                        // Mutating requests block other requests from starting to ensure an up to date snapshot is used.
-                        // Since we're explicitly awaiting exceptions to mutating requests will bubble up here.
-                        await WrapStartRequestTaskAsync(work.StartRequestAsync(context, handler, cancellationToken), rethrowExceptions: true).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        // Non mutating are fire-and-forget because they are by definition read-only. Any errors
-                        // will be sent back to the client but they can also be captured via HandleNonMutatingRequestError,
-                        // though these errors don't put us into a bad state as far as the rest of the queue goes.
-                        // Furthermore we use Task.Run here to protect ourselves against synchronous execution of work
-                        // blocking the request queue for longer periods of time (it enforces parallelizability).
-                        var currentWorkTask = WrapStartRequestTaskAsync(Task.Run(() => work.StartRequestAsync(context, handler, cancellationToken), cancellationToken), rethrowExceptions: false);
-
-                        if (CancelInProgressWorkUponMutatingRequest)
-                        {
-                            if (currentWorkCts is null)
-                            {
-                                throw new InvalidOperationException($"unexpected null value for {nameof(currentWorkCts)}");
-                            }
-
-                            if (!concurrentlyExecutingTasks.TryAdd(currentWorkTask, currentWorkCts))
-                            {
-                                throw new InvalidOperationException($"unable to add {nameof(currentWorkTask)} into {nameof(concurrentlyExecutingTasks)}");
-                            }
-
-                            _ = currentWorkTask.ContinueWith(t =>
-                            {
-                                if (!concurrentlyExecutingTasks.TryRemove(t, out var concurrentlyExecutingTaskCts))
-                                {
-                                    throw new InvalidOperationException($"unexpected failure to remove task from {nameof(concurrentlyExecutingTasks)}");
-                                }
-
-                                concurrentlyExecutingTaskCts.Dispose();
-                            }, CancellationToken.None, TaskContinuationOptions.ExecuteSynchronously, TaskScheduler.Default);
-                        }
-                    }
+                    // Determine which language is appropriate for handling the request.
```

Comment:
flesh this out.   let's be specific about how LSP + DidOpen works and how this gets the semantics we waznt.